File: Vigenere/Vigenere.py
#!/usr/bin/env python3
#Vigenere Cipher
class message:
    def __init__(self, message, key):
        self.__msg = message
        self.__keyPhrase = key
        self.__encryption = self.__encrypt()
        self.__decryption = self.__decrypt()
    def get_encryption(self):
        return self.__encryption
    def get_decryption(self):
        return self.__decryption
    def __encrypt(self):
        alphabet = ' ABCDEFGHIJKLMNOPQRSTUVWXYZ'
        mesg = self.__msg.upper()
        cipher = ""
        i = 0
        for letter in mesg:
            index = (alphabet.find(letter) + (alphabet.find(self.__keyPhrase[i]))) % len(alphabet)
            cipher += alphabet[index]
            i += 1
            if i == len(self.__keyPhrase):
                i = 0
        return cipher
    def __decrypt(self):
        alphabet = ' ABCDEFGHIJKLMNOPQRSTUVWXYZ'
        msg = self.__msg.upper()
        plainText = ""
        i = 0
        for letter in msg:
            index = (alphabet.find(letter) - (alphabet.find(self.__keyPhrase[i]))) % len(alphabet)
            plainText += alphabet[index]
            i += 1
            if i == len(self.__keyPhrase):
                i = 0
        return plainText

while(True):
    mode = input("Would you like to encrypt, decrypt, or quit? Enter 'e', 'd', or 'q': ")
    if mode == 'e':
        msg = input("Enter the message to encrypt: ")
        key = input("Enter the key: ")
        key = key.upper()
        code = message(msg, key)
        print("\n\nEncrypted Message: "+code.get_encryption()+"\n\n")
    elif mode == 'd':
        msg = input("Enter the message to decrypt: ")
        key = input("Enter the key: ")
        key = key.upper()
        code = message(msg, key)
        print("\n\nDecrypted Message: "+code.get_decryption()+"\n\n")
    elif mode == 'q':
        print("\nGoodbye.")
        break
    else:
        continue

File: Vigenere/Vigenere.java
import java.util.Scanner;
public class Vigenere
{
	public static final String LETTERS = " ABCDEFGHIJKLMNOPQRSTUVWXYZ";
	public static void main(String[] args) 
	{
		Scanner s = new Scanner(System.in);
		String mode, msg;
		String key;
		while(true)
		{
			System.out.println("Would you like to encrypt, decrypt, or quit? Enter 'e', 'd', or 'q': ");
			mode = s.next();
			if (mode.equalsIgnoreCase("e"))
			{
				s.nextLine();
				System.out.println("Enter the message to encrypt: ");
				msg = s.nextLine();
				System.out.println("Enter the key: ");
				key = s.nextLine();
				System.out.println("\n\nCipher Text: "+encrypt(msg, key)+"\n\n");
			}
			else if (mode.equalsIgnoreCase("d"))
			{
				s.nextLine();
				System.out.println("Enter the message to decrypt: ");
				msg = s.nextLine();
				System.out.println("Enter the key: ");
				key = s.nextLine();
				System.out.println("\n\nPlain Text: "+decrypt(msg, key)+"\n\n");
			}
			else if (mode.equalsIgnoreCase("q")) {
				System.out.println("Goodbye.");
				System.exit(0);
			}
			else
				continue;
		}
	}
	public static String encrypt(String message, String key)
	{
		String cipher = "";
		String msg = message.toUpperCase();
		String phrase = key.toUpperCase();
		int index;
		int keyIndex = 0;
		for (int i = 0; i < msg.length(); i++)
		{
			index = (LETTERS.indexOf(msg.charAt(i)) + LETTERS.indexOf(phrase.charAt(keyIndex))) % LETTERS.length();
			cipher += Character.toString(LETTERS.charAt(index));
			keyIndex++;
			if (keyIndex == key.length())
				keyIndex = 0;
		}
		return cipher;
	}
	public static String decrypt(String message, String key)
	{
		String plainText = "";
		String msg = message.toUpperCase();
		String phrase = key.toUpperCase();
		int index;
		int keyIndex = 0;
		for (int i = 0; i < msg.length(); i++)
		{
			index = (LETTERS.indexOf(msg.charAt(i)) - LETTERS.indexOf(phrase.charAt(keyIndex))) % LETTERS.length();
			plainText += Character.toString(LETTERS.charAt(index));
			keyIndex++;
			if (keyIndex == key.length())
				keyIndex = 0;
		}
		return plainText;
	}
}
